CoreMapView/src/au/gov/asd/tac/constellation/views/mapview2/markers/PolygonMarker.java

```diff
@@ -0,0 +1,155 @@
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package au.gov.asd.tac.constellation.views.mapview2.markers;
+
+import au.gov.asd.tac.constellation.views.mapview2.MapView;
+import java.util.ArrayList;
+import java.util.List;
+import javafx.event.EventHandler;
+import javafx.scene.input.MouseEvent;
+import javafx.scene.paint.Color;
+import javafx.scene.shape.Line;
+
+/**
+ * Marker that represent the shapes the user draws
+ *
+ * @author altair1673
+ */
+public class PolygonMarker extends AbstractMarker {
+
+    private final List<Line> polygonLineUI = new ArrayList<>();
+    private Line currentLine = null;
+    private String rawPath = "";
+
+    public PolygonMarker(final MapView parent, final int markerID, final int xOffset, final int yOffset) {
+        super(parent, markerID, -99, xOffset, yOffset, AbstractMarker.MarkerType.POLYGON_MARKER);
+
+        markerPath.setStroke(Color.BLACK);
+        markerPath.setFill(Color.ORANGE);
+        markerPath.setOpacity(0.4);
+
+        // Event handler for the polygon marker
+        markerPath.setOnMouseEntered(new EventHandler<MouseEvent>() {
+            public void handle(final MouseEvent e) {
+
+                markerPath.setFill(Color.YELLOW);
+
+                e.consume();
+            }
+        });
+
+        markerPath.setOnMouseExited(new EventHandler<MouseEvent>() {
+            public void handle(final MouseEvent e) {
+
+                markerPath.setFill(Color.ORANGE);
+
+                e.consume();
+            }
+        });
+
+        markerPath.setOnMouseClicked(new EventHandler<MouseEvent>() {
+            public void handle(final MouseEvent e) {
+
+                parent.removeUserMarker(markerID);
+                e.consume();
+            }
+        });
+    }
+
+    /**
+     * Adds a new line to the polygon marker
+     *
+     * @param prevLineEndX
+     * @param prevLineEndY
+     * @return
+     */
+    public Line addNewLine(final double prevLineEndX, final double prevLineEndY) {
+        // If current line is null it mean the first line is being drawn
+        // Else finish off the current line and add it to the UI
+        if (currentLine == null) {
+            currentLine = new Line();
+            currentLine.setStartX(prevLineEndX);
+            currentLine.setStartY(prevLineEndY);
+            currentLine.setEndX(prevLineEndX);
+            currentLine.setEndY(prevLineEndY);
+        } else {
+            currentLine.setEndX(prevLineEndX);
+            currentLine.setEndY(prevLineEndY);
+            polygonLineUI.add(currentLine);
```

Comment:
seems like this is the only part of the code that needs to be in a if block, the rest of the if and else code blocks are identical